### PR TITLE
Fix deadlock

### PR DESCRIPTION
**Pull request by LuKks**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
   "dependencies": {
     "b4a": "^1.6.1",
     "mutexify": "^1.4.0",
-    "streamx": "^2.12.4"
+    "streamx": "^2.13.2"
```
Comment (LuKks) [inline package.json:27]: Libraries like `hyperdrive-next` that uses `hyperblobs` are using latest `streamx`, so the deadlock appears on `hyperdrive-next`

File: lib/streams.js [hunk 2]
```diff
@@ -57,13 +57,26 @@ class BlobWriteStream extends Writable {
...
   }
 
+  _destroy (cb) {
+    if (this._release) {
+      this._release()
+      this._release = null
+    }
+
+    cb(null)
+  }
```
Comment (LuKks) [inline lib/streams.js:78]: Unsure if this is the most precise fix but seems ok

File: lib/streams.js [hunk 3]
```diff
@@ -57,13 +57,26 @@ class BlobWriteStream extends Writable {
   _final (cb) {
     this._append(err => {
       if (err) return cb(err)
```
Comment (LuKks) [inline lib/streams.js:59]: Not doing lock release here because this `cb(err)` ends up calling `_destroy`

**Comment by LuKks** [comment]
I was trying to fix a bug in `mirror-drive` that leaded me to this

**Comment by LuKks** [comment]
@mafintosh Ready!

File: lib/streams.js (hunk 2 above)
Comment (mafintosh) [reply lib/streams.js:78]: _destroy is always called, so just release it there always and remove the releasing in _final

File: lib/streams.js [hunk 4]
```diff
@@ -59,11 +59,15 @@ class BlobWriteStream extends Writable {
...
       this.id.blockLength = this.core.length - this.id.blockOffset
       this.id.byteLength = this.core.byteLength - this.id.byteOffset
-      this._release()
       return cb(null)
     })
   }
 
+  _destroy (cb) {
+    if (this._release) this._release()
```
Comment (mafintosh) [inline lib/streams.js:67]: release is always set no?

Comment (LuKks) [reply lib/streams.js:67]: It's set on streamx `_open`, after core being ready:
https://github.com/holepunchto/hyperblobs/blob/baf4766abffa7f4cdda22f93751fc8493a585e51/lib/streams.js#L49

I think `_open` is not always called?

Comment (mafintosh) [reply lib/streams.js:67]: 👍

**Pull request merged**